modules/Bio/EnsEMBL/Compara/DBSQL/PeptideAlignFeatureAdaptor.pm

```diff
@@ -577,39 +612,75 @@ sub fetch_all_by_dbID_list {
-  my $self             = shift;
-  my $qmember_id       = shift;
-  my $hit_genome_db_id = shift;
+sub fetch_BRH_by_member_genomedb {
+    my ($self, $qmember_id, $hit_genome_db_id) = @_;
+
+    return unless($qmember_id and $hit_genome_db_id);
+    my $member = $self->db->get_SeqMemberAdaptor->fetch_by_dbID($qmember_id);
+
+    # using trick of specifying table twice so can join to self
+    my $extrajoin;
+    my $constraint = "paf.hit_rank=1 AND paf2.hit_rank=1 AND paf.qmember_id=? AND paf.hgenome_db_id=?";
+    $self->{_curr_gdb_id} = $member->genome_db_id;
+    # dependent on whether peptide_align_feature is a table specific to query/hit genome
+    if ( table_exists( $self->dbc, 'peptide_align_feature_' . $hit_genome_db_id ) ) {
+        $extrajoin = [
+            [ ['peptide_align_feature_' . $hit_genome_db_id, 'paf2'],
+            'paf.qmember_id=paf2.hmember_id AND paf.hmember_id=paf2.qmember_id',
+            {'paf2.peptide_align_feature_id AS pafid2' => '_rhit_dbID'} ]
+        ];
+    }
+    # peptide_align_feature is a single table shared amongst one or more genomes
+    else {
+        $extrajoin = [
+            [ ['peptide_align_feature', 'paf2'],
+            'paf.qmember_id=paf2.hmember_id AND paf.hmember_id=paf2.qmember_id',
+            {'paf2.peptide_align_feature_id AS pafid2' => '_rhit_dbID'} ]
+        ];
+    }
+    $self->bind_param_generic_fetch($qmember_id, SQL_INTEGER);
+    $self->bind_param_generic_fetch($hit_genome_db_id, SQL_INTEGER);
+    # exact match duplicates acceptable, we want all the BRHs
+    return $self->generic_fetch($constraint, $extrajoin);
 
-  #print(STDERR "fetch_all_RH_by_member_genomedb qmember_id=$qmember_id, genome_db_id=$hit_genome_db_id\n");
-  return unless($qmember_id and $hit_genome_db_id);
+}
 
-  my $member = $self->db->get_SeqMemberAdaptor->fetch_by_dbID($qmember_id);
 
-  $self->{_curr_gdb_id} = $member->genome_db_id;
+=head2 fetch_BBH_by_member_genomedb
 
-   my $extrajoin = [
-                     [ ['peptide_align_feature_'.$hit_genome_db_id, 'paf2'],
-                       'paf.qmember_id=paf2.hmember_id AND paf.hmember_id=paf2.qmember_id',
-                       {'paf2.peptide_align_feature_id AS pafid2' => '_rhit_dbID'}]
-                   ];
+  Arg [1]    : seq_member_id of query peptide member
+  Arg [2]    : genome_db_id of hit species
+  Example    : $paf = $adaptor->fetch_BBH_by_member_genomedb(31957, 3);
+  Description: Returns the top ranked PeptideAlignFeature from the database
+  Returntype : array reference of Bio::EnsEMBL::Compara::PeptideAlignFeature objects
+  Exceptions : none
+  Caller     : general
 
-   my $constraint = "paf.hit_rank=1 AND paf2.hit_rank=1 AND paf.qmember_id=? AND paf.hgenome_db_id=?";
-  $self->bind_param_generic_fetch($qmember_id, SQL_INTEGER);
-  $self->bind_param_generic_fetch($hit_genome_db_id, SQL_INTEGER);
+=cut
+
+
+sub fetch_BBH_by_member_genomedb {
+    my ($self, $qmember_id, $hit_genome_db_id) = @_;
```

Comment:
I would prefer to keep it as it is (for all), because that is the name of the `peptide_align_feature` column, so in a way it is easier to understand and link back I think